kview/src/main/resources/dev/ikm/komet/kview/mvvm/view/details/concept-details.fxml

```diff
@@ -72,16 +72,20 @@
                               <SVGPath content="M10.4946 5.39299C9.15593 5.39299 8.09879 6.44558 8.09879 7.78882C8.09879 8.60528 8.46989 9.32176 9.05295 10.0351C9.63089 10.7422 10.4513 11.4861 11.4073 12.353L11.4145 12.3595L12.1821 13.0583L12.9503 12.3591L12.9569 12.353C13.913 11.4861 14.7334 10.7422 15.3113 10.0351C15.8944 9.32176 16.2655 8.60528 16.2655 7.78882C16.2655 6.44558 15.2083 5.39299 13.8696 5.39299C13.2412 5.39299 12.6377 5.63586 12.1821 6.03418C11.7266 5.63586 11.1231 5.39299 10.4946 5.39299ZM9.51546 7.78882C9.51546 7.22291 9.92872 6.80965 10.4946 6.80965C10.9425 6.80965 11.3767 7.10748 11.5228 7.48235L11.6055 7.69465H12.7587L12.8415 7.48235C12.9876 7.10748 13.4217 6.80965 13.8696 6.80965C14.4355 6.80965 14.8488 7.22291 14.8488 7.78882C14.8488 8.20766 14.6208 8.66806 14.117 9.25677C13.6512 9.80109 12.9934 10.4064 12.1804 11.1433C11.3691 10.4064 10.7123 9.8012 10.2469 9.25688C9.74346 8.66811 9.51546 8.2077 9.51546 7.78882Z  M4.9165 1.66699C3.53579 1.66699 2.4165 2.78628 2.4165 4.16699V15.8337C2.4165 17.2144 3.53579 18.3337 4.9165 18.3337H16.5832C17.9639 18.3337 19.0832 17.2144 19.0832 15.8337V4.16699C19.0832 2.78628 17.9639 1.66699 16.5832 1.66699H4.9165ZM4.08317 4.16699C4.08317 3.70675 4.45627 3.33366 4.9165 3.33366H16.5832C17.0434 3.33366 17.4165 3.70675 17.4165 4.16699V15.8337C17.4165 16.2939 17.0434 16.667 16.5832 16.667H4.9165C4.45627 16.667 4.08317 16.2939 4.08317 15.8337V4.16699Z" fillRule="EVEN_ODD" styleClass="favorite-concept" />
                            </graphic>
                         </Button>
-                        <ToggleButton fx:id="reasonerToggleButton" mnemonicParsing="false" onAction="#openReasonerSlideout" styleClass="reasoner" text="ToggleButton">
+                        <Button fx:id="reasonerToggleButton" layoutX="96.0" layoutY="16.0" mnemonicParsing="false" onAction="#popupAddContextMenu" styleClass="reasoner" text="Button">
+                           <contextMenu>
+                              <ContextMenu fx:id="reasonerContextMenu" opacity="0.99" styleClass="kview-context-menu">
+                                 <items>
+                                    <MenuItem mnemonicParsing="false" text="Run full reasoner" />
+                                    <MenuItem mnemonicParsing="false" text="Run incremental reasoner" />
+                                    <MenuItem mnemonicParsing="false" text="Redo navigation" />
```

Comment:
(Optional) Can you add functions that get called for each menu item? 
My reasoning: Whoever is assigned the implementation ticket could focus on the controller class.

Reply:
yes... will do...

Reply:
updated